### DISCUSSION TITLE
How to define a custom edge component as a default one.

### DISCUSSION THREAD

**Discussion by tpoxa** [bcakmakoglu/vue-flow#129] [Q&A] [answered]
I use a custom edge component like in examples with the delete button in the middle. It works well but how to make that component be applied when I connect nodes by drag and drop... Found only `default-edge-options`, was looking for `default-edge-component`

Thanks!

**Comment by bcakmakoglu** [comment, marked as answer]
The edge component that is looked for when setting an edge is determined by the `edge.type`

So you probably call something like this:
```ts
onConnect((params) => {
    // will fall back to default edge type as none is specified
    addEdges([params])
})
```

You can either specify the edge type at this point or define a default edge type by settings `defaultEdgeOptions`

So either you do it like this
```ts
onConnect((params) => {
    // will create an edge of type `custom-edge`
    addEdges([
      {
        type: 'custom-edge'
        ...params,
      }
    ])
})
```

or like this
```vue
<!-- *every* new edge will be of type `custom-edge` -->
<VueFlow :default-edge-options="{ type: 'custom-edge' }" />
```

Either way you have to specify the type of your created edge.
Afterwards it's the same as using custom nodes, either provide a component to the `edgeTypes` parameter or use a slot with the name of your edge-type.

**Reply by bcakmakoglu** [reply]
Just for more context,
`onConnect` provides `params` which consists of `source`, `target`, `sourceHandleId`, `targetHandleId`
everything else is up to you to fill in, any [Edge](https://types.vueflow.dev/interfaces/Edge.html) properties are valid.

**Reply by tpoxa** [reply]
Thank you!
